Casks/citrix-workspace.rb

```diff
@@ -1,25 +1,28 @@
 cask "citrix-workspace" do
-  version "23.04.0.36,2304"
-  sha256 :no_check
+  arch arm: "Universal", intel: "Intel"
 
-  url "https://downloadplugins.citrix.com/Mac/CitrixWorkspaceApp.dmg"
+  version "23.04.0.36"
+  sha256 arm:   "8f24b593d3095464af27ddfc85d7b19d38f48d88f2a8afa31694591b00c3cdcf",
+         intel: "29a6267509af5b3ade006799136829d9ab12c32491a192c1941db290a68e5990"
+
+  url "https://downloadplugins.citrix.com/ReceiverUpdates/Prod/Receiver/Mac/CitrixWorkspaceApp#{arch}#{version}.pkg"
   name "Citrix Workspace"
   desc "Managed desktop virtualization solution"
   homepage "https://www.citrix.com/"
 
   livecheck do
-    url "https://www.citrix.com/downloads/workspace-app/mac/workspace-app-for-mac-latest.html"
+    url "https://downloadplugins.citrix.com/ReceiverUpdates/Prod/catalog_macos2.xml"
     strategy :page_match do |page|
-      match = page.match(/Version:*\s+(\d+(?:\.\d+)+)\s\((\d+(?:\.\d+)*)\)/i)
+      match = page.match(%r{<Installer platformArchitecture="#{arch}">.*?<Version>([\d.]+)</Version>}m)
       next if match.blank?
 
-      "#{match[1]},#{match[2]}"
+      match[1]
     end
   end
```

Comment:
This might be a bit simpler. Matching on `ShortDescription` as `HDX RealTime Media Engine` is included in this file.
```suggestion
  livecheck do
    url "https://downloadplugins.citrix.com/ReceiverUpdates/Prod/catalog_macos2.xml"
    regex(%r{<short\S*?>.*?workspace.*?</short\S*?>.*?<version>v?(\d+(?:\.\d+)+)</version>}i)
  end
```

Reply:
Sure, although for RTME, the platform is `Any` currently.

Reply:
Good catch!